### ISSUE TITLE
lastResult.deref is not a function (it is undefined)

### ISSUE THREAD

**Issue by quocluongha** [reduxjs/reselect#669]
I updated to v5 and I encountered this error message when using `createSelectorCreator` with `weakMapMemoize`.

I returned a number in `selectFormPosition` and this error happened, but I tried to change the returned result into object then it is working fine.

The `createDeepEqualSelector` function:
```js
export const createDeepEqualSelector = createSelectorCreator(weakMapMemoize, {
  resultEqualityCheck: isEqual, // lodash isEqual
});
```

The state:
```js
const initialState = {
  formPosition: 0,
};
```

The selector functions:
```js
export const selectWithdrawalState = (state: RootState) =>
  state.shopper.profile.withdrawal;

export const selectFormPosition = createDeepEqualSelector(
  selectWithdrawalState,
  state => state.formPosition,
  // state => ({ formPosition: state.formPosition }), <-- this is working fine

);
```

![image](https://github.com/reduxjs/reselect/assets/80971410/c38d6cc0-1c66-4082-8be0-d1d0dd7da263)

**Comment by quocluongha** [comment]
Thank you @aryaemami59, that is my temporary solution up to now.

**Comment by aryaemami59** [comment]
@quocluongha Just a quick question, are you using React Native or Expo?

**Comment by quocluongha** [comment]
@aryaemami59 I am using React Native 0.72.7

**Comment by aryaemami59** [comment]
@quocluongha Thank you, I'll put up a PR with the fix soon.

**Comment by EskiMojo14** [comment]
Fix released in [v5.1.0](https://github.com/reduxjs/reselect/releases/tag/v5.1.0)
